indexd/index/drivers/query/urls.py

```diff
@@ -80,13 +84,37 @@ def query_urls(
                 query = query.having(
                     ~q_func["string_agg"](IndexRecordUrl.url, ",").contains(exclude)
                 )
+
+            # add authz filter
+            if any_authz:
+                # if any_authz is set, we want to filter records that have ANY of the authz elements
+                # was getting the following warning
+                #  SAWarning: SELECT statement has a cartesian product between FROM element(s) "index_record" and FROM element "index_record_url".  Apply join condition(s) between each element to resolve. .all()
+                # This warning means the query is selecting from multiple tables (index_record and index_record_url) without a proper join condition, causing a cartesian product. To fix this, so we explicitly join IndexRecordUrl to IndexRecord using their relationship.
+                # originally it was:
+                # Add an explicit join between IndexRecordUrl and IndexRecord
+                sub = session.query(IndexRecordAuthz.did).filter(
+                    IndexRecordAuthz.resource.in_(any_authz)
+                )
+                query = query.filter(IndexRecord.did.in_(sub.with_entities(IndexRecordAuthz.did).statement))
+
+
             # [('did', 'urls')]
+
             record_list = (
                 query.order_by(IndexRecordUrl.did.asc())
                 .offset(offset)
                 .limit(limit)
                 .all()
             )
+            # alternatively, we could have used the following query:
```

Comment:
let's just remove this unless there's a reason to switch to it in the future

Reply:
done